src/Reflection/ReflectionProperty.php

```diff
@@ -592,9 +671,123 @@ private function computeModifiers(PropertyNode $node): int
         $modifiers  = $node->isReadonly() ? ReflectionPropertyAdapter::IS_READONLY : 0;
         $modifiers += $node->isStatic() ? CoreReflectionProperty::IS_STATIC : 0;
         $modifiers += $node->isPrivate() ? CoreReflectionProperty::IS_PRIVATE : 0;
+        $modifiers += $node->isPrivateSet() ? ReflectionPropertyAdapter::IS_PRIVATE_SET_COMPATIBILITY : 0;
         $modifiers += $node->isProtected() ? CoreReflectionProperty::IS_PROTECTED : 0;
+        $modifiers += $node->isProtectedSet() ? ReflectionPropertyAdapter::IS_PROTECTED_SET_COMPATIBILITY : 0;
         $modifiers += $node->isPublic() ? CoreReflectionProperty::IS_PUBLIC : 0;
+        $modifiers += ($node->flags & ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY) === ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY ? ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY : 0;
+        $modifiers += ($node->flags & Modifiers::ABSTRACT) === Modifiers::ABSTRACT ? ReflectionPropertyAdapter::IS_ABSTRACT_COMPATIBILITY : 0;
 
+        /** @phpstan-ignore return.type */
         return $modifiers;
     }
+
+    private function computeImmediateVirtual(PropertyNode $node): bool
+    {
+        if ($node->hooks === []) {
+            return false;
+        }
+
+        $setHook = null;
+
+        foreach ($node->hooks as $hook) {
+            if ($hook->name->name === 'set') {
+                $setHook = $hook;
+                break;
+            }
+        }
+
+        if ($setHook === null) {
+            return true;
+        }
+
+        $setHookBody = $setHook->getStmts();
+        assert($setHookBody !== null);
+
+        // Short syntax
+        if (count($setHookBody) === 1 && $setHookBody[0] instanceof Node\Stmt\Return_) {
+            return false;
+        }
+
+        $visitor   = new FindingVisitor(static fn (Node $node): bool => $node instanceof Node\Expr\Assign);
+        $traverser = new NodeTraverser($visitor);
+        $traverser->traverse($setHookBody);
+
+        foreach ($visitor->getFoundNodes() as $assigNode) {
```

Comment:
Overall, the logic works, but any code including closures that use `$this` on another object may trip it.

Reply:
Yes, I can make it more complicated to ignore $this in anonymous classes, but I’n not sure if it will be enough.